### PR TITLE
txn: Move tag and cmd_metric to each command

### PR DESCRIPTION
**Pull request by longfangsong**
Signed-off-by: longfangsong <longfangsong@icloud.com>

### What problem does this PR solve?

Minor refactor in `src/txn/commands.rs`

### What is changed and how it works?

What's Changed:
Move tag and cmd_metric to each command and abstract them with a `CommandExt` trait. 

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: src/storage/txn/commands.rs [hunk 1]
```diff
@@ -511,12 +526,21 @@ impl ScanLock {
     }
 }
 
+impl CommandExt for ScanLock {
```
Comment (longfangsong) [inline src/storage/txn/commands.rs:529]: PTAL #8099.

**Comment by longfangsong** [comment]
@nrc PTAL

File: src/storage/txn/commands.rs [hunk 2]
```diff
@@ -240,10 +240,15 @@ impl From<MvccGetByStartTsRequest> for TypedCommand<Option<(Key, MvccInfo)>> {
...
+pub trait CommandExt {
+    fn tag(&self) -> metrics::CommandKind;
+    fn incr_cmd_metric(&self);
+}
+
 macro_rules! command {
     (
         $(#[$outer_doc: meta])*
-        $cmd: ident -> $cmd_ty: ty {
+        $cmd: ident, $cmd_ty: ty, $tag_and_cmd_metric: ident {
```
Comment (nrc) [inline src/storage/txn/commands.rs:251]: Why change `->` to `,`?

Comment (nrc) [reply src/storage/txn/commands.rs:251]: Also I think just using `tag` rather than `tag_and_cmd_metric` is clearer

Comment (longfangsong) [reply src/storage/txn/commands.rs:251]: > Why change `->` to `,`?

I think `Prewrite -> Vec<Result<()>>, prewrite` syntax is a little bit ambiguous when reading. 
Is it 
`(Prewrite -> Vec<Result<()>>), prewrite`
or 
`Prewrite -> (Vec<Result<()>>, prewrite)`
?

But I'm not so sure about what's the proper syntax here, especially there would be more things to put here (eg. `ts`) in the future, what do you have in mind?

Comment (nrc) [reply src/storage/txn/commands.rs:251]: Ah, it is a good point! I think just using commas is fine. There is no proper syntax here, just whatever is easiest to read.

**Pull request merged**